### DISCUSSION TITLE
`ValueError: PointArray does not have the same number of rows as skeleton nodes`

### DISCUSSION THREAD

**Discussion by ab-madabhushi** [talmolab/sleap#1208] [Help!] [answered]
Hello,

I have trained a model using 6 videos to track different body parts of a spider and was trying to use this model to predict instances on a new set of videos (30 new videos). For this, I added to the 'new' videos to the .slp file with the model and asked it predict instances on all frames across all the videos. I see this(screenshot below) pop-up ibut, then I do not see any predictions nor does it go to a new video to try and predict. I see this error in the terminal:
"PointArray does not have the same number of rows as skeleton "
ValueError: PointArray does not have the same number of rows as skeleton nodes.

I successfully predicted all the frames for the 6 videos on which the model was trained but I seem to run into this issue after I added new videos and I do not know what changed. Could you please suggest what I ma doing wrong. Thanks!
![image](https://user-images.githubusercontent.com/116664143/222520973-67c15757-ec38-400d-9bea-12b2427346e0.png)

Thanks

**Comment by roomrys** [comment, marked as answer]
Hi @ab-madabhushi,

Whoa, that error looks like it will only pop-up if we are predicting a different `Skeleton` than the one contained in our project, i.e. our model skeleton does not match our project skeleton. We can verify this by looking at data.labels.skeletons section in the training config json and checking if the skeleton there matches the skeleton in our project (under the Skeleton tab in the GUI).

Let us know if that is the case, otherwise, we can help troubleshoot further!

Thanks,
Liezl

**Reply by roomrys** [reply]
Hi @ab-madabhushi,

Just following up to see if you have had any luck in identify whether there were two separate skeletons being used? If you are still having trouble, you can send you data (slp, videos, models) to lmaree@salk.edu for troubleshooting.

Thanks,
Liezl

**Reply by ab-madabhushi** [reply]
Hello,

So sorry to not have replied earlier. Yes, it was using a different model, more specifically it appears that SLEAP uses the latest model present in the folder to predict instances. I am currently running multiple training protocols i.e, training SLEAP to track different species of jumping spiders parallely so, I was wondering if there was a smart way to ensure that the correct model is used to predict instances instead of keeping track of timestamps and changing the model manually before inferring predictions.

Would be really grateful for your suggestions!

Thanks,
Abhinav

**Reply by roomrys** [reply]
Hi @ab-madabhushi,

If you have models for both species in the same `models` folder, then SLEAP will get confused and could potentially pull up the wrong model (for the species at hand). SLEAP attempts to automatically load the latest trained model in the adjacent `model` folder; thus, if you have mixed models in the `models` folder it is likely that you might get the wrong model loaded. To rectify this, we recommend separating the projects into their own respective directories (one directory per species).

Let us know if this helps!

Thanks,
Liezl
